lib/parallel_tests/grouper.rb

```diff
@@ -46,6 +51,47 @@ def in_even_groups_by_size(items, num_groups, options = {})
 
       private
 
+      def specify_groups(items, num_groups, options, groups)
+        specify_test_process_groups = options[:specify_groups].split('|')
+        if specify_test_process_groups.count > num_groups
+          raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+        end
+
+        all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+        specified_items_found, items = items.partition { |item, _size| all_specified_tests.include?(item) }
+
+        specified_specs_not_found = all_specified_tests - specified_items_found.map(&:first)
+        if specified_specs_not_found.any?
+          raise "Could not find #{specified_specs_not_found} from --specify-groups in the main selected files & folders"
+        end
+
+        if specify_test_process_groups.count == num_groups && items.flatten.any?
```

Comment:
do we need the flatten ?
```suggestion
        if specify_test_process_groups.count == num_groups && items.any?
```

Reply:
I think we do, sorry, I accidentally resolved the convo we had before https://github.com/grosser/parallel_tests/pull/797#discussion_r572352941

Reply:
ah yeah, the empty groups issue 😞